### DISCUSSION TITLE
Trying to get watermeter into Home Assistant

### DISCUSSION THREAD

**Discussion by robertendicott** [jomjol/AI-on-the-edge-device#2074] [General] [answered]
I am using the following:
AIO version: 14.0.3
Meter: Digital meter (Sensus - iPerl)

I am not having any issues with image recognition, but on the Overview page only the RAW Value gets updated.  The Value: is blank and the Previous Value: is at the default 42.0 and does not change.  Why would the Value not get updated?
![01-Digitizer](https://user-images.githubusercontent.com/16157730/221262982-83c7ee0f-ee37-44b0-8f12-dff95506d6ed.png)

I have MQTT setup on Home Assistant and have setup the AIO configuration to use MQTT and the Homeassistnat Discovery option.  I just want to make sure that this config looks correct.
![03-Digitizer](https://user-images.githubusercontent.com/16157730/221263599-ed8e87e1-1a9c-42e5-8846-e3062b9060fd.png)

Once I go to Home Assistant and the water meter in MQTT there is no information on the Sensors but there is in Diagnostic and the Raw value is correct.  This does not seem correct to me.
![06-Digitizer](https://user-images.githubusercontent.com/16157730/221264078-07512720-c1fd-4ab2-a128-973475f06be5.png)

The last thing is when I go to the Energy Dashboard  under Water Consumption there is nothing to select for Water usage.
![07-Digitizer](https://user-images.githubusercontent.com/16157730/221265648-90947480-9e1e-4f0c-9e1b-95162f7731fc.png)

I know this is a lot of questions/issue but I really want this project to work as I have tried  the clamps that go over the water pipe and use "ultrasonic rays" to track the water but it failed miserably and is really expensive.  This is a really awesome project and thank you for your help.

**Comment by caco3** [comment]
> Why would the Value not get updated?

See the error message: `Rate to high`. You have to set the Pre-Value correctly: Menu `Settings > Set Previous Value`. And also the Parameter https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-numbermaxratevalue might need adjustment.

> Once I go to Home Assistant and the water meter in MQTT there is no information on the Sensors but there is in Diagnostic and the Raw value is correct. This does not seem correct to me.

That is to be expected and should be fixed once you fix above issue!

> The last thing is when I go to the Energy Dashboard under Water Consumption there is nothing to select for Water usage.

can you try (just for testing), if it works if you use `m3` instead of `gal`? My (older HA) says, it must be either `m3` or `gl`. How ever according to your screenshot, it expects now `gal` instead.

**Comment by robertendicott** [comment]
Thank you so much for the reply, you pointed me in the right direction.   I have been trying to read the documentation and the discussions for the past week but it wasn't sinking in.  I just changed the MaxRateValue to 10 as I have the Auto Timer Interval at 3 minutes.  Having 8 people in the house can use a lot of water.

The Home Assistant and the water meter in MQTT are correct now.

For the Configure Water Consumption I tried setting m3 instead of gal and I'm still seeing "No matching statistics found"

But at least now I have progress.

**Reply by caco3** [reply]
maybe you have to wait a bit, it takes up to 2h until it shows the consumption in the energy meter.

**Comment by robertendicott** [comment, marked as answer]
Just a quick update, there must of been something wrong with the HA MQTT.  I created a new HA iVM and configured MQTT and pointed AI on the edge  to the new instance and it worked fine.  It was detectable right away.

![09-Digitizer](https://user-images.githubusercontent.com/16157730/221712093-c773ad37-4197-40d4-b89a-b1bb5f1558df.png)

Thanks for the help and pointing me in the right direction.
